### PR TITLE
Add bazel buildsystem support

### PR DESCRIPTION
**Pull request by mjcarroll**
This introduces necessary infrastructure and changes to make ign-common build with Bazel.

### CODE REVIEW

File: .github/ci-bionic/packages.apt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+pkg-config
```
Comment (chapulina) [inline .github/ci-bionic/packages.apt:1]: The file is already under `.github/ci`

https://github.com/ignitionrobotics/ign-common/blob/ign-common3/.github/ci/packages.apt

File: graphics/src/ColladaExporter.cc [hunk 2]
```diff
@@ -41,24 +41,10 @@ static void LogTinyXml2DocumentError(
   {
     warning += std::to_string(_doc.ErrorID()) + "): ";
 
-#ifdef TINYXML2_MAJOR_VERSION_GE_6
```
Comment (chapulina) [inline graphics/src/ColladaExporter.cc:44]: Have you checked the TinyXML2 versions on all our supported platforms?

**Review by chapulina — COMMENTED** [review]
Same comment here about separating the Bazel addition from the other changes.

**Review by JShep1 — COMMENTED** [review]
Looks good, but I'm getting a test failure in `PluginLoader` [here](https://github.com/ignitionrobotics/ign-common/blob/ign-common3/src/PluginLoader_TEST.cc#L55-L58).  Are you getting any failures on your end?

**Comment by mjcarroll** [comment]
> Looks good, but I'm getting a test failure in PluginLoader here. Are you getting any failures on your end?

Fixed

File: graphics/src/ColladaExporter.cc (hunk 2 above)
Comment (mjcarroll) [reply graphics/src/ColladaExporter.cc]: I reverted this.

**Review by JShep1 — CHANGES_REQUESTED** [review]
All is working for me locally except for `av`, I'm getting some error messages on Bionic for missing headers in `ffmpeg`: `hdr_dynamic_metadata.h` and `ac3_parser.h`.  I was thinking this could possibly be a versioning thing, I searched `ffmpeg` in my `/usr/include` directory and couldn't find those two files even though I'm fully updated locally.

Additionally, I'm getting a build failure with `colcon`, but changing `test/util.hh` to `test_util.hh` [here](https://github.com/ignitionrobotics/ign-common/blob/bazel-common3/test/integration/encoder_timing.cc#L21) fixes that.

File: graphics/src/ColladaExporter.cc [hunk 3]
```diff
@@ -25,6 +25,11 @@
 
 #include "tinyxml2.h"
 
+#if TINYXML2_MAJOR_VERSION >= 6
```
[the remark this answers is not in the document]
Comment (JShep1) [reply graphics/src/ColladaExporter.cc:28]: I replicated both above issues on my Bionic desktop and both.  Removing:
```
 - libavcodec/ac3_parser.h
 - libavcodec/adts_parser.h
 - libavutil/hdr_dynamic_metadata.h
 - libavutil/encryption_info.h
 - libavutil/hwcontext_mediacodec.h
 - libavutil/tx.h
```
from the file [here](https://github.com/ignitionrobotics/ign-bazel/blob/master/third_party/ffmpeg.BUILD) fixed the `ffmpeg` issue for me.  The second issue was also resolved by me defining that tinyxml2 variable.  All tests passing after I did those steps, again, not sure how we actually want to define that variable without hardcoding the version number.

**Review by chapulina — APPROVED** [review]
I only checked the non-Bazel parts and they look reasonable to me :+1:

**Pull request merged**